### PR TITLE
Monday

### PR DESCRIPTION
**Pull request by RochQA**

### CODE REVIEW

File: src/main/java/com/qa/trainer/controller/TrainerController.java [hunk 1]
```diff
@@ -1,18 +1,51 @@
...
+		this.rest = rest;
+		this.client = client;
+	}
+	
+	@PutMapping("/checkValid")
+	public String checkTrainer(@RequestBody CreateAccount createAccount) {
+		return srvc.checkTrainer(createAccount, getAllTrainers());
+	}
+	@PutMapping("/checkUpdateValid")
+	public String checkUpdateValid(@RequestBody UpdateAccount account) {
```
Comment (JHarry444) [inline src/main/java/com/qa/trainer/controller/TrainerController.java:39]: Paths should be constants.

File: src/main/java/com/qa/trainer/service/TrainerServiceImpl.java [hunk 2]
```diff
@@ -1,15 +1,54 @@
...
+import com.qa.trainer.entities.Trainer;
+import com.qa.trainer.entities.UpdateAccount;
+
 @Service
 public class TrainerServiceImpl implements TrainerService {
-
+	
+	
+	public String checkDuplicates(CreateAccount accountC, List<Trainer> trainers) {
+		Trainer matchingTrains = new Trainer();
```
Comment (JHarry444) [inline src/main/java/com/qa/trainer/service/TrainerServiceImpl.java]:
```suggestion
		Trainer trainer = new Trainer();
```

File: src/test/java/com/example/demo/TrainerServiceTests.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+import com.qa.trainer.entities.Trainer;
+import com.qa.trainer.entities.UpdateAccount;
+import com.qa.trainer.service.TrainerServiceImpl;
+
+@RunWith(MockitoJUnitRunner.class)
+@SpringBootTest(classes = TrainerApplication.class)
+public class TrainerServiceTests {
+	
+	Trainer jordan;
+	Trainer chester;
```
Comment (JHarry444) [inline src/test/java/com/example/demo/TrainerServiceTests.java]: Encapsulation

**Review by JHarry444 — CHANGES_REQUESTED** [review]

File: src/main/java/com/qa/trainer/service/TrainerServiceImpl.java (hunk 2 above)
Comment (RochQA) [reply src/main/java/com/qa/trainer/service/TrainerServiceImpl.java]: changed to matchingTrainer as trainer is used inside the stream

**Pull request merged**